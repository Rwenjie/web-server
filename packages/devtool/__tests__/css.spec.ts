import { resolveFixture, runWebpack } from "./test-utils";
import generateCssLoaders from "../lib/config/css";
import MiniCssExtractPlugin from "mini-css-extract-plugin";

it("should support css imports", async () => {
	const output = await runWebpack({
		mode: "development",
		entry: resolveFixture("style.css"),
		devtool: false,
		output: {
			path: "/",
		},
		module: {
			rules: generateCssLoaders({
				production: false,
				extract: true,
				sourceMap: false,
			}),
		},
		plugins: [
			new MiniCssExtractPlugin({ filename: "[name].css" }),
		],
	});

	expect(output.readFileSync("/main.css", "utf8")).toMatchSnapshot();
});

it('should support less', async () => {
	const output = await runWebpack({
		mode: "development",
		entry: resolveFixture("less.less"),
		devtool: false,
		output: {
			path: "/",
		},
		module: {
			rules: generateCssLoaders({
				production: false,
				extract: true,
				sourceMap: false,
			}),
		},
		plugins: [
			new MiniCssExtractPlugin({ filename: "[name].css" }),
		],
	});

	expect(output.readFileSync("/main.css", "utf8")).toMatchSnapshot();
});
