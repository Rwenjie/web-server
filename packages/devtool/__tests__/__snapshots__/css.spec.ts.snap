// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should support css imports 1`] = `
"/* This file should be import by another */

#app::before {
	display: flex;
}

.test {
	transition: .5s;
}

"
`;

exports[`should support less 1`] = `
"/* This file should be import by another */
#app::before {
  display: flex;
}

.parent.child {
  width: 20rem;
}

"
`;
